packages/flutter_tools/lib/src/commands/build_ios.dart

```diff
@@ -19,24 +19,9 @@ import 'build.dart';
 /// Builds an .app for an iOS app to be used for local testing on an iOS device
 /// or simulator. Can only be run on a macOS host. For producing deployment
 /// .ipas, see https://flutter.dev/docs/deployment/ios.
-class BuildIOSCommand extends BuildSubCommand {
-  BuildIOSCommand({ @required bool verboseHelp }) {
-    addTreeShakeIconsFlag();
-    addSplitDebugInfoOption();
+class BuildIOSCommand extends _BuildIOSSubCommand {
+  BuildIOSCommand({ @required bool verboseHelp }) : super(verboseHelp: verboseHelp) {
     addBuildModeFlags(defaultToRelease: true);
```

Comment:
Archive doesn't need build mode flags, it's always `release`.

Reply:
There are no use cases for a profile archive? or does that sort of defeat the purpose

Reply:
You could want a non-release archive, but the use case is pretty rare.  I can add the flags in and default to release, can't hurt...
https://github.com/flutter/flutter/issues/57557

Reply:
Done.